### PR TITLE
[JAVA][Rest-assured] Added constants

### PR DESCRIPTION
**Pull request by viclovsky**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [ ] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.0.x`. Default: `master`.
- [ ] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR
Added constant "httpMethod", "summary" and fixed javadoc for "returnType", "dataType".

### CODE REVIEW

**Comment by wing328** [comment]
@viclovsky thanks for the PR.

cc technical committee:
@bbdouglas (2017/07) @JFCote (2017/08) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01)

File: modules/openapi-generator/src/main/resources/Java/libraries/rest-assured/api.mustache [hunk 1]
```diff
@@ -85,7 +85,9 @@ public class {{classname}} {
     {{/isDeprecated}}
     public class {{operationIdCamelCase}}Oper {
 
+        public static final String REQ_METHOD = "{{httpMethod}}";
         public static final String REQ_URI = "{{path}}";
+        public static final String SUMMARY = "{{{summary}}}";
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/Java/libraries/rest-assured/api.mustache]: @viclovsky what's the use case for this constant storing the summary (which is included in the javadoc string above)?

Comment (viclovsky) [reply modules/openapi-generator/src/main/resources/Java/libraries/rest-assured/api.mustache]: Constants are necessary to fill report. 
Rest-assured is mostly client for testing. Tests often has integration with test report (such as [Allure](https://github.com/allure-framework/allure2)). Test report has detailed information about tests run results, curl arguments, command line etc.

Comment (wing328) [reply modules/openapi-generator/src/main/resources/Java/libraries/rest-assured/api.mustache]: @viclovsky thanks for the explanation 👍

Comment (viclovsky) [reply modules/openapi-generator/src/main/resources/Java/libraries/rest-assured/api.mustache]: It also intended to do the same with tags, but I haven't found easy way to do it.

**Comment by wing328** [comment]
UPDATE: I'm looking into the issue and trying to fix the CI (which should report red instead)

**Comment by wing328** [comment]
Performed some more tests and CI (CircleCI) was able to report syntax errors with the Java rest-assured client. Still not entirely sure why the Javadoc error was missed (and I was able to repeat the Javadoc error locally)

**Comment by jeff9finger** [comment]
I saw those changes to the javadoc, but did not think they should cause any issues. I guess the java doc generation is where things break. Is that right?

**Comment by wing328** [comment]
Here is the error I got locally:
```
[ERROR] /Users/williamcheng/Code/openapi-generator/samples/client/petstore/java/rest-assured/src/main/java/org/openapitools/client/api/UserApi.java:271: error: unknown tag: User
[ERROR]          * @param user (List<User>) List of user object (required)
[ERROR]                             ^
[ERROR]
[ERROR] Command line was: /Library/Java/JavaVirtualMachines/jdk1.8.0_172.jdk/Contents/Home/jre/../bin/javadoc @options @packages
[ERROR]
[ERROR] Refer to the generated Javadoc files in '/Users/williamcheng/Code/openapi-generator/samples/client/petstore/java/rest-assured/target/apidocs' dir.
[ERROR]
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```
We suspect CircleCI failed to detect the error due to an old version of JDK8 used in the build environment.

**Pull request merged**